Managers/ConsoleMan.cpp

```diff
@@ -1,541 +1,423 @@
+			SetConsoleScreenSize(m_ConsoleScreenRatio - 0.05F);
+		}
+
+		if (!m_ReadOnly) {
+			m_InputTextBox->SetEnabled(true);
+			m_InputTextBox->SetFocus();
+	
+			// Get rid of any grave accents (`) that are pasted/typed into the textbox.
+			std::string textBoxString = m_InputTextBox->GetText();
+			if (std::find(textBoxString.begin(), textBoxString.end(), '`') != textBoxString.end()) {
+				textBoxString.erase(std::remove(textBoxString.begin(), textBoxString.end(), '`'), textBoxString.end());
+				m_InputTextBox->SetText(textBoxString);
+				m_InputTextBox->SetCursorPos(textBoxString.length());
+			}	
+		} else {
+			m_InputTextBox->SetEnabled(false);
+			m_GUIController->GetManager()->SetFocus(0);
+			return;
+		}	
+
+		if (g_UInputMan.KeyPressed(KEY_ENTER) || g_UInputMan.KeyPressed(KEY_ENTER_PAD)) { FeedString(); }
+
+		// If multiple strings were pasted into the box, we need to check for and submit all the strings
+		if (m_InputTextBox->GetText().find_last_of('\n') != std::string::npos) { FeedMultipleStrings(); }
+
+		if (!m_InputLog.empty() && !g_UInputMan.FlagCtrlState()) {
+			if (g_UInputMan.KeyPressed(KEY_UP)) {
+				LoadLoggedInput(false);
+			} else if (g_UInputMan.KeyPressed(KEY_DOWN)) {
+				LoadLoggedInput(true);
+			}
+		}
+
+		// TODO: Get this working and see if it actually makes any difference.
+		/*
+		// Cut off the text in the text label at a reasonable height so it doesn't get really slow to draw
+		if (m_ConsoleText->GetTextHeight() > g_FrameMan.GetResY() {
+		}
+		*/
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::FeedString() {
+		// Clear the errors from the VM so we will get fresh feedback
+		g_LuaMan.ClearErrors();
+
+		// Add the input line to the console
+		m_ConsoleText->SetText(m_ConsoleText->GetText() + "\n" + m_InputTextBox->GetText());
+		g_LuaMan.RunScriptString(m_InputTextBox->GetText(), false);
+
+		// Add any error feedback from the lua VM
+		if (!g_LuaMan.ErrorExists()) { m_ConsoleText->SetText(m_ConsoleText->GetText() + "\n" + "ERROR: " + g_LuaMan.GetLastError()); }
+
+		// Save the input made to the log, but only if it's different from the last one
+		if (!m_InputTextBox->GetText().empty() && (m_InputLog.empty() || m_InputLog.front().compare(m_InputTextBox->GetText()) != 0)) { m_InputLog.push_front(m_InputTextBox->GetText()); }
+
+		// Reset the log marker
+		m_InputLogPosition = m_InputLog.begin();
+		m_LastLogMove = 0;
+
+		// Clear the input line
+		m_InputTextBox->SetText("");
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::FeedMultipleStrings() {
+		char strLine[1024];
+		std::stringstream inputStream(m_InputTextBox->GetText());
+
+		while (!inputStream.fail()) {			
+			inputStream.getline(strLine, 1024, '\n');
+			std::string line = strLine;
+			
+			if (line != "\r" && !line.empty()) {
```

Comment:
Seems like you could call FeedString here with a little tweaking (mainly inputting the string as a parameter and having another flag to not ClearErrors at the start).

Also, should it not clear the input text box at the end?